misk/src/test/kotlin/misk/moshi/wire/WireMessageAdapterTest.kt

```diff
@@ -0,0 +1,171 @@
+package misk.moshi.wire
+
+import com.google.inject.util.Modules
+import com.squareup.moshi.Moshi
+import com.squareup.protos.test.parsing.Shipment
+import com.squareup.protos.test.parsing.Warehouse
+import misk.moshi.MoshiModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import okio.ByteString
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@MiskTest(startService = false)
+internal class WireMessageAdapterTest {
+    @MiskTestModule
+    val module = Modules.combine(MoshiModule())
+
+    @Inject
+    lateinit var moshi: Moshi
+
+    @Test
+    fun simpleTypes() {
+        val warehouseAdapter = moshi.adapter(Warehouse::class.java)
+
+        val warehouse = Warehouse.Builder()
+                .warehouse_id(1014L)
+                .warehouse_token("AAAA")
+                .build()
+
+        assertThat(warehouseAdapter.toJson(warehouse))
+                .isEqualTo("""{"warehouse_id":1014,"warehouse_token":"AAAA"}""")
+        assertThat(warehouseAdapter.fromJson("""{"warehouse_id":1014,"warehouse_token":"AAAA"}"""))
+                .isEqualTo(warehouse)
+    }
+
+    @Test
+    fun nestedTypes() {
+        val shipmentAdapter = moshi.adapter(Shipment::class.java)
+
+        val shipment = Shipment.Builder()
+                .shipment_id(100075)
+                .shipment_token("P_AAAAA")
+                .status(Shipment.State.DELIVERING)
+                .source(Warehouse.Builder()
+                        .warehouse_id(9999L)
+                        .warehouse_token("C_RANDY")
+                        .build())
+                .destination(Warehouse.Builder()
+                        .warehouse_id(7777L)
+                        .warehouse_token("C_CATHY")
+                        .build())
+                .deleted(true)
+                .load_ratio(0.75)
+                .notes(listOf("Note A", "Note B", "Note C"))
+                .build()
+
+        val jsonText = shipmentAdapter.indent(" ").toJson(shipment)
+        assertThat(jsonText).isEqualTo("""
```

Comment:
You can use trimPrefix() or whatever if you'd like. 

Example:
https://github.com/square/kotlinpoet/blob/master/src/test/java/com/squareup/kotlinpoet/KotlinPoetTest.kt

Reply:
Works especially good with the IntelliJ feature where you can drag a bunch of cursors to edit multiple lines at once.

Reply:
I used ``trimIndent``, but it's still finicky. Added an [``isEqualToAsJson``](https://github.com/square/misk/pull/102/files#diff-32dad8de370a7758a2282dfacab4f3cdR11) method instead which normalizes the non-quoted whitespace before comparing the two strings.